### DISCUSSION TITLE
Dynamic Tooltip for an action button

### DISCUSSION THREAD

**Discussion by gopalwebdev** [Power-Components/livewire-powergrid#488] [Q&A] [answered]
Hello, I'm looking for a feature in actureRules() fn where we can add tooltip dynamically. Eg:     

      public function actionRules(): array
          {
              return [
                  Rule::button('toggleBillStatus')
                  ->when(fn($model) => $model->is_billed)
                  ->setAttribute('class', 'fas fa-check-circle text-green p-2 border-0 bg-transparent')
                  ->setAttribute('tooltip', 'Billed'),
      
                  Rule::button('toggleBillStatus')
                  ->when(fn($model) => !$model->is_billed)
                  ->setAttribute('class', 'fas fa-check-circle text-dark p-2 border-0 bg-transparent')
                  ->setAttribute('tooltip', 'Unbilled'),
              ];
          }

**Comment by d2x** [comment, marked as answer]
You could probably do it with [microtip](https://github.com/ghosh/microtip).

```
Rule::button('toggleBillStatus')
              ->when(fn($model) => !$model->is_billed)
              ->setAttribute('class', 'fas fa-check-circle text-dark p-2 border-0 bg-transparent')
              ->setAttribute('aria-label', 'Unbilled'),
              ->setAttribute('role', 'tooltip'),
              ->setAttribute('data-microtip-position', 'top'),
```

**Comment by luanfreitasdev** [comment]
This is excellent, I believe it solves the issue. 

Thanks!
